src/components/Viewer/Panel/Qualification.jsx

```diff
@@ -32,19 +40,19 @@ const Qualification = ({ file = {}, t, f, lang }) => {
   } = metadata
   const { label } = qualification
 
-  // TODO Improve it when other contact choices are needed
-  useEffect(() => {
-    let isMounted = true
-    ;(async () => {
-      const contactCollection = client.collection('io.cozy.contacts')
-      const { data: currentUser } = await contactCollection.findMyself()
-      isMounted && setCurrentUser(currentUser[0])
-    })()
+  const contactIds = getReferencedBy(file, 'io.cozy.contacts').map(
+    contact => contact.id
+  )
+  const contactByIdsQuery = getContactByIds(contactIds)
+  const { data: contactList } = useQuery(
+    contactByIdsQuery.definition,
+    contactByIdsQuery.options
+  )
 
-    return () => {
-      isMounted = false
-    }
-  }, [client])
+  const contactsDisplayed =
+    contactList?.map((contact, idx) => {
+      return `${contact.fullname}${idx !== contactList.length - 1 ? ', ' : ''}`
```

Comment:
Be careful you should not assume that all contacts are well formatted. So to get the fullname you should use an helper method: https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/models/contact.js#L121